xbmc/utils/GLUtils.h

```diff
@@ -8,25 +8,21 @@
 
 #pragma once
 
-// GL Error checking macro
-// this function is useful for tracking down GL errors, which otherwise
-// just result in undefined behavior and can be difficult to track down.
-//
-// Just call it 'VerifyGLState()' after a sequence of GL calls
-//
-// if GL_DEBUGGING and HAS_GL are defined, the function checks
-// for GL errors and prints the current state of the various matrices;
-// if not it's just an empty inline stub, and thus won't affect performance
-// and will be optimized out.
+/*
+ GL Error checking macro
+ this function is useful for tracking down GL errors, which otherwise
+ just result in undefined behavior and can be difficult to track down.
+
+ Just call it 'VerifyGLState()' after a sequence of GL calls
+
+ OpenGL component logging must be enabled
+ */
 
 #include "system_gl.h"
 
 void _VerifyGLState(const char* szfile, const char* szfunction, int lineno);
-#if defined(GL_DEBUGGING) && (defined(HAS_GL) || defined(HAS_GLES))
```

Comment:
I don't think enabling this always is a good idea. VerifyGLState is called really often. Only checking the setting that often generates a lot of useless load.

Reply:
So what is the alternative? OpenGL debug code that is never used?

I'm not sure if there is a callback anywhere that we can set instead of constantly checking the setting.

Reply:
IMO it did exactly what it was supposed to do. Those who compiled with GL_DEBUGGING did get the debug logs. Personally I never did, I set breakpoints on the right places and used mesa debug capabilities.

Reply:
> IMO it did exactly what it was supposed to do. Those who compiled with GL_DEBUGGING did get the debug logs. Personally I never did, I set breakpoints on the right places and used mesa debug capabilities.

Sure, but how many people even knew about defining `GL_DEBUGGING`? This provides a way for everyone to have OpenGL error checking and is much more simple for those who don't have much OpenGL knowledge (read: me).

Reply:
compiling with GL_DEBUGGING has nothing to do with OpenGL knowledge. "everyone" does not need this debug output because "everyone" does not know how to interpret the output nor how to fix issues.
On the other hand "everyone" suffers from wasting CPU cycles. I seriously doubt that this debug code will reveal any issues once release has been shipped. Hence I see no justification for this additional load.

Thats my view on it but I won't fight for it. Go ahead if you feel this is a required improvement. (I am lucky because I know how to safe my systems from that load)

Reply:
With @FernetMenta here. `glGetError` is just not intended for release builds. 

(Admittedly unreliable) sources even claim it's slow on Android - https://gamedev.stackexchange.com/questions/54646/calling-glgeterror-in-release-builds

Reply:
Yep, totally fair points. I'll take another look at things and see what I can come up with